libraries/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/ws/GraphQLWsProtocol.kt

```diff
@@ -147,22 +168,75 @@ class GraphQLWsProtocol(
   /**
    * A factory for [GraphQLWsProtocol].
    *
-   * @param connectionPayload a map of additional parameters to send in the "connection_init" message
-   * @param pingIntervalMillis the interval between two client-initiated pings or -1 to not send any ping.
-   * Default value: -1
-   * @param pingPayload the ping payload to send in "ping" messages or null to not send a payload
-   * @param pongPayload the pong payload to send in "pong" messages or null to not send a payload
-   * @param connectionAcknowledgeTimeoutMs the timeout for receiving the "connection_ack" message, in milliseconds
-   * @param frameType the type of the websocket frames to use. Default value: [WsFrameType.Text]
    */
-  class Factory(
-      private val connectionPayload: suspend () -> Map<String, Any?>? = { null },
-      private val pingIntervalMillis: Long = -1,
-      private val pingPayload: Map<String, Any?>? = null,
-      private val pongPayload: Map<String, Any?>? = null,
-      private val connectionAcknowledgeTimeoutMs: Long = 10_000,
-      private val frameType: WsFrameType = WsFrameType.Text,
-  ) : WsProtocol.Factory {
+  class Factory constructor() : WsProtocol.Factory {
```

Comment:
I'm considering renaming `WsProtocol.Factory` to `WsProtocol.Builder` at some point since they more or less serve the same purpose